site-manage/site-manage-tool/tool/src/java/org/sakaiproject/site/tool/SiteAction.java

```diff
@@ -9749,215 +9762,262 @@ private void actionForTemplate(String direction, int index,
+									return;
+								}
 
-												if (StringUtils.isNotBlank(toolId) && !notStealthOrHiddenTool(toolId)) {
-													// Found a stealthed tool, queue for removal
-													log.debug("found stealthed tool {}", toolId);
-													rmToolList.add(toolConf);
+								// SAK-20797
+								long oldSiteQuota = this.getSiteSpecificQuota(oldSiteId);
+
+								// Create the duplicate site
+								Site site = SiteService.addSite(newSiteId, getStateSite(state));
+								site.setTitle(titleStripped);
+
+								// If the site contains the Email Archive tool, we're going to check for valid/unique email address;
+								// this requires the duplicate site to already exist...
+								if (state.getAttribute(STATE_DUP_SITE_HAS_EMAIL_ARCHIVE) != null) {
+									String newEmailID = StringUtils.trimToNull(params.getString("emailAddress"));
+									if (StringUtils.isBlank(newEmailID)) {
+										addAlert(state, rb.getString("java.emailarchive"));
+										deleteTempDupSiteOnError(site);
+									} else {
+										state.setAttribute(STATE_TOOL_EMAIL_ADDRESS, newEmailID);
+										if (StringUtils.isNotBlank(newEmailID)) {
+											String channelReference = mailArchiveChannelReference(site.getId());
+											if (!Validator.checkEmailLocal(newEmailID)) {
+												addAlert(state, rb.getString("java.theemail"));
+												deleteTempDupSiteOnError(site);
+											} else if (!aliasService.allowSetAlias(newEmailID, channelReference)) {
+												addAlert(state, rb.getString("java.addalias"));
+												deleteTempDupSiteOnError(site);
+											} else {
+												try {
+													// First clear any alias set to the channel
+													aliasService.removeTargetAliases(channelReference);
+
+													// Check to see whether the alias has been used
+													String target = aliasService.getTarget(newEmailID);
+													boolean targetsThisSite = site.getReference().equals(target);
+													if (!targetsThisSite) {
+														addAlert(state, rb.getFormattedMessage("java.emailinuse", new Object[]{newEmailID, ServerConfigurationService.getServerName()}));
+														deleteTempDupSiteOnError(site);
+													}
+												} catch (IdUnusedException ex) {
+													// If aliasService.getTarget() throws this, it's all good: email alias not yet in use, so we set it
+													try {
+														aliasService.setAlias(newEmailID, channelReference);
+													} catch (Exception e) {
+														addAlert(state, rb.getFormattedMessage("unexpectedError", new Object[] {ServerConfigurationService.getString("mail.support")}));
+														deleteTempDupSiteOnError(site);
+													}
+												} catch (Exception ex) {
+													addAlert(state, rb.getFormattedMessage("unexpectedError", new Object[] {ServerConfigurationService.getString("mail.support")}));
+													deleteTempDupSiteOnError(site);
```

Comment:
ends here, with the exception of the new `deleteTempDupSiteOnError()` defined below